bundles/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/ISYActiveBinding.java

```diff
@@ -0,0 +1,718 @@
+			final UDNode node, final Command command) {
+
+		// Do percent first so dimmers dim and do not just go on and off. 
+		if(command instanceof IncreaseDecreaseType && ISYNodeType.DIMMER.equals(config.getType())) {
+			IncreaseDecreaseType type = (IncreaseDecreaseType) command;
+			
+			switch (type) {
+			case INCREASE:
+				if (node instanceof UDGroup) {
+					this.insteonClient.brightenScene(node.address);
+				} else {
+					this.insteonClient.brightenDevice(node.address);
+				}
+				break;
+			case DECREASE:
+				if (node instanceof UDGroup) {
+					this.insteonClient.dimScene(node.address);
+				} else {
+					this.insteonClient.dimDevice(node.address);
+				}
+				break;
+			}
+		} else if (command instanceof OnOffType) {
+			OnOffType type = (OnOffType) command;
+
+			switch (type) {
+			case ON:
+				if (ISYNodeType.THERMOSTAT.equals(config.getType())) {
+					// turn thermostat Fan State to On
+					this.insteonClient.changeNodeState(ISYControl.CLIFS.name(),
+							"7", node.address);
+
+					// provide immediate feedback such that the UI gets updated
+					// or else the switch button will not be refreshed until the
+					// new state has been reached
+					this.eventPublisher.postUpdate(config.getItemName(), type);
+				} else if (ISYNodeType.LOCK.equals(config.getType())) {
+				    this.insteonClient.changeNodeState(ISYControl.SECMD.name(), 
+				            "1", node.address);
+				    this.eventPublisher.postUpdate(config.getItemName(), type);
+				} else if (node instanceof UDGroup) {
+					this.insteonClient.turnSceneOn(node.address);
+				} else {
+					// TODO: See if openhab can handle a slow or fast or make it
+					// a config option.
+					this.insteonClient.turnDeviceFastOn(node.address);
+				}
+				break;
+
+			case OFF:
+				if (ISYNodeType.THERMOSTAT.equals(config.getType())) {
+					// turn thermostat off
+					this.insteonClient.changeNodeState(ISYControl.CLIMD.name(),
+							"0", node.address);
+					// provide immediate feedback such that the UI gets updated
+					// or else the switch button will not be refreshed until the
+					// new state has been reached
+					this.eventPublisher.postUpdate(config.getItemName(), type);
+				
+                } else if (ISYNodeType.LOCK.equals(config.getType())) {
+                    this.insteonClient.changeNodeState(ISYControl.SECMD.name(), 
+                            "0", node.address);
+                    this.eventPublisher.postUpdate(config.getItemName(), type);
+				} else if (node instanceof UDGroup) {
+					this.insteonClient.turnSceneFastOff(node.address);
+				} else {
+					this.insteonClient.turnDeviceFastOff(node.address);
+				}
+				break;
+			}
+		} else if (command instanceof DecimalType) {
+			DecimalType type = (DecimalType) command;
+
+			switch (config.getType()) {	
+			case THERMOSTAT:
+				switch (config.getControlCommand()) {
+				case CLISPH:
+				case CLISPC:
+					DecimalType value = new DecimalType(
+							type.doubleValue() * 2.0);
```

Comment:
Recommend using BigDecimal math instead of native floating point to avoid imprecise results.